modos/api.py

```diff
@@ -473,28 +474,39 @@ def from_file(
             )
 
         instance_list = []
+        modo_dict = {}
         for element in element_list:
             metadata = element.get("element")
             args = element.get("args", {})
             if metadata.get("@type") == "MODO":
                 del metadata["@type"]
-                modo = cls(
-                    path=object_directory,
-                    s3_endpoint=s3_endpoint,
-                    s3_kwargs=s3_kwargs or {"anon": True},
-                    htsget_endpoint=htsget_endpoint,
-                    **metadata,
-                    **args,
-                )
+                modo_dict["meta"] = metadata
+                modo_dict["args"] = args
             else:
                 metadata = set_data_path(metadata, args.get("source_file"))
                 inst = dict_to_instance(metadata)
                 instance_list.append((inst, args))
 
+        modo = cls(
+            path=object_directory,
+            s3_endpoint=s3_endpoint,
+            s3_kwargs=s3_kwargs or {"anon": True},
+            htsget_endpoint=htsget_endpoint,
+            **modo_dict.get("meta", {}),
+            **modo_dict.get("args", {}),
+        )
+
         modo_ids = {Path(id).name: id for id in modo.metadata.keys()}
         for inst, args in instance_list:
             if inst.id in modo_ids.keys():
                 modo.update_element(modo_ids[inst.id], inst)
             else:
                 modo.add_element(inst, **args)
+        if not no_remove:
+            modo_id = modo_dict.get("meta", {}).get("id") or modo.path.name
```

Comment:
**question:** could we not always get `id` from `modo`?
**suggestion** use `modo_id = modo.zarr['/'].attrs['id']`